ert/data/record/_transformation.py

```diff
@@ -22,71 +24,80 @@ def _prepare_location(base_path: Path, location: Path) -> None:
-        self,
-        transmitter: RecordTransmitter,
-        mime: str,
-        runpath: Path,
-        location: Path,
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
         _prepare_location(runpath, location)
-        await transmitter.dump(runpath / location, mime)
+        await _save_record(record, runpath / location, mime)
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        await transmitter.transmit_file(runpath / location, mime)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return await _load_record(location, mime)
 
 
 class TarRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        record = await transmitter.load()
         if isinstance(record, BlobRecord):
             with tarfile.open(fileobj=io.BytesIO(record.data), mode="r") as tar:
                 _prepare_location(runpath, location)
                 tar.extractall(runpath / location)
         else:
             raise TypeError("Record needs to be a BlobRecord type!")
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        blob_record = BlobRecord(data=await make_tar(runpath / location))
-        await transmitter.transmit_record(blob_record)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return BlobRecord(data=await make_tar(location))
 
 
 class ExecutableRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        # pre-make bin folder if neccessary
+        # pre-make bin folder if necessary
         base_path = Path(runpath / _BIN_FOLDER)
         base_path.mkdir(parents=True, exist_ok=True)
 
         # create file(s)
         _prepare_location(base_path, location)
-        await transmitter.dump(base_path / location, mime)
+        await _save_record(record, base_path / location, mime)
 
-        # post-proccess if neccessary
+        # post-proccess if necessary
         path = base_path / location
         st = path.stat()
         path.chmod(st.st_mode | stat.S_IEXEC)
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        await transmitter.transmit_file(runpath / location, mime)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return await _load_record(location, mime)
+
+
+async def _load_record(
+    file: Path,
```

Comment:
I find it hard to guess what "`load"`" refers to, the code reveals that it is loading from disk, and not loading from some storage f.ex. Can we make this more explicit in either the function name or a docstring? Maybe `_load_record_from_file`?